test/integration/apiserver/flowcontrol/fight_test.go

```diff
@@ -0,0 +1,192 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flowcontrol
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"math/rand"
+	"sync"
+	"testing"
+	"time"
+
+	flowcontrol "k8s.io/api/flowcontrol/v1beta1"
+	"k8s.io/apimachinery/pkg/util/clock"
+	genericfeatures "k8s.io/apiserver/pkg/features"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	utilfc "k8s.io/apiserver/pkg/util/flowcontrol"
+	fqtesting "k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/testing"
+	"k8s.io/apiserver/pkg/util/flowcontrol/metrics"
+	"k8s.io/client-go/informers"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	featuregatetesting "k8s.io/component-base/featuregate/testing"
+)
+
+/* fightTest configures a test of how API Priority and Fairness config
+   controllers fight when they disagree on how to set FlowSchemaStatus.
+   In particular, they set the condition that indicates integrity of
+   the reference to the PriorityLevelConfiguration.  The scenario tested is
+   two teams of controllers, where the controllers in one team set the
+   condition normally and the controllers in the other team set the condition
+   to the opposite value.
+
+   This is a behavioral test: it instantiates these controllers and runs them
+   almost normally.  The test aims to run the controllers for a little under
+   2 minutes.  The test takes clock readings to get upper and lower bounds on
+   how long each controller ran, and calculates consequent bounds on the number
+   of writes that should happen to each FlowSchemaStatus.  The test creates
+   an informer to observe the writes.  The calculated lower bound on the
+   number of writes is very lax, assuming only that one write can be done
+   every 10 seconds.
+*/
+type fightTest struct {
+	t              *testing.T
+	ctx            context.Context
+	loopbackConfig *rest.Config
+	teamSize       int
+	stopCh         chan struct{}
+	now            time.Time
+	clk            *clock.FakeClock
+	ctlrs          map[bool][]utilfc.Interface
+
+	countsMutex sync.Mutex
+
+	// writeCounts maps FlowSchema.Name to number of writes
+	writeCounts map[string]int
+}
+
+func newFightTest(t *testing.T, loopbackConfig *rest.Config, teamSize int) *fightTest {
+	now := time.Now()
+	ft := &fightTest{
+		t:              t,
+		ctx:            context.Background(),
+		loopbackConfig: loopbackConfig,
+		teamSize:       teamSize,
+		stopCh:         make(chan struct{}),
+		now:            now,
+		clk:            clock.NewFakeClock(now),
```

Comment:
do we use this fake clock anywhere?  If not, that means we don't need it for the testableconfig either, right?

Reply:
Oops, I forgot to remove it here.
I have a bit of a preference for keeping the injection style in the regular code.

Reply:
Fixed in #100061